mne/io/boxy/boxy.py

```diff
@@ -0,0 +1,301 @@
+# Authors: Kyle Mathewson, Jonathan Kuziek <kuziek@ualberta.ca>
+#
+# License: BSD (3-clause)
+
+import re as re
+
+import numpy as np
+
+from ..base import BaseRaw
+from ..meas_info import create_info
+from ...utils import logger, verbose, fill_doc
+from ...annotations import Annotations
+
+
+@fill_doc
+def read_raw_boxy(fname, datatype='AC', preload=False, verbose=None):
+    """Reader for a BOXY optical imaging recording.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the BOXY data file.
+    datatype : str
+        Type of data to return (AC, DC, or Ph).
+    %(preload)s
+    %(verbose)s
+
+    Returns
+    -------
+    raw : instance of RawBOXY
+        A Raw object containing BOXY data.
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+    return RawBOXY(fname, datatype, preload, verbose)
+
+
+@fill_doc
+class RawBOXY(BaseRaw):
+    """Raw object from a BOXY optical imaging file.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the BOXY data file.
+    datatype : str
+        Type of data to return (AC, DC, or Ph).
+    %(preload)s
+    %(verbose)s
+
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+
+    @verbose
+    def __init__(self, fname, datatype='AC', preload=False, verbose=None):
+        logger.info('Loading %s' % fname)
+
+        # Determine which data type to return.
+        if datatype not in ['AC', 'DC', 'Ph']:
+            raise RuntimeError('Expect AC, DC, or Ph, got %s' % datatype)
+
+        # Read header file and grab some info.
+        filetype = 'parsed'
+        start_line = 0
+        end_line = 0
+        mrk_col = 0
+        mrk_data = list()
+        col_names = list()
+        with open(fname, 'r') as data:
```

Comment:
Can we test the boxy version number is supported?

Reply:
good idea

Reply:
I just want to clarify...

Did you want to us to grab the boxy version from the raw data file, save this somewhere in the raw data object, and...

a) confirm that it's a version we know is supported and/or...
b) confirm that it's the version we expect?

Either way, I assume the boxy version number should be stored somewhere in `.info`?

Reply:
I was thinking something like...

https://github.com/mne-tools/mne-python/blob/master/mne/io/nirx/nirx.py#L110

I assume the file format for boxy has change over the 80+ versions. So we want to make sure that the user is passing in data that you can actually parse.

Reply:
Okay perfect, I can add this. Thanks for the help.